### ISSUE TITLE
[Issue]: Seasons and Specials showing when files not present

### ISSUE THREAD

**Issue by Brramble** [jellyfin/jellyfin#12208]
### Please describe your bug

Since updating to 10.9.7 from 10.9.6, I have noticed a bug where some shows will now show the "Specials" season when there are no specials available for that show. Please see example below:
![image](https://github.com/jellyfin/jellyfin/assets/21260984/16309472-59ba-47e2-ac69-556e08b92c28)

Some shows, for example "Love Island", is showing me ALL seasons when I only have the most recent downloaded. All other seasons are empty and there are no files to suggest that Jellyfin needs to show and populate them. Please see example below:
![image](https://github.com/jellyfin/jellyfin/assets/21260984/c2193508-09f3-477b-b9f3-85febf59e083)

### Reproduction Steps

When running 10.9.7, check shows for specials if enabled.

When also running 10.9.7, check shows which have multiple seasons, but only a few downloaded locally.

### Jellyfin Version

10.9.0

### Environment

```markdown
- OS: Slackware 15.0
- Linux Kernel: 6.1.79-Unraid
- Virtualization: Docker (LinuxServer Image)
- Clients: Andriod, WebOS / Client, Swiftfin.
- Browser: Firefox & Chrome
- FFmpeg Version: latest
- Playback Method: N/A
- Hardware Acceleration: N/A
- GPU Model: N/A
- Plugins: Default + TMDB, Intro Skipper, Infuse Sync
- Reverse Proxy: Yes
- Base URL: N/A
- Networking: N/a
- Storage: N/A
```

### Jellyfin logs

```shell
No logs can be found to connect to this bug.
```

### FFmpeg logs

```shell
No logs can be found to connect to this bug.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by JamsRepos** [comment]
It's important to note, refreshing all metadata fixes the issue. However, on a full library scan the same issue persists. 

**When refreshing metadata**
![image](https://github.com/jellyfin/jellyfin/assets/1347620/c38b8b44-d62b-4598-b4e4-c6feddb34a18)
**After a library scan**
![image](https://github.com/jellyfin/jellyfin/assets/1347620/5138e062-37f1-41be-9b5b-64ee27ddb5dc)

**Comment by Brramble** [comment]
@Shadowghost - was this something you mentioned in the Discord about your latest changes to seasons?

**Comment by Shadowghost** [comment]
Yes. Somewhere is a bug creating unnecessary seasons (or more precisely not cleaning up temporary seasons correctly). But I'm too busy to debug and fix it right now.

**Comment by fernvenue** [comment]
By the way, even if I lock the show, it can still be reproduced and may display non-existent seasons.

**Comment by Travel2Here** [comment]
i found this issue too.posted in discussion channel.need fix~~

**Comment by TheMelmacian** [comment]
It seems it has something to do with the "Missing Episode Fetcher" metadata provider. After deactivating it in the show library, the empty seasons don't reappear after deletion even if I trigger a rescan of the affected show.

To delete the unwanted seasons create the season folders of the unwanted entries and rescan the show. Afterwards delete them via the GUI.
Deleting them directly doesn't work for me. The GUI shows an error message but I can't find anything in the logs.

According to Travel2Here, it also helps to deactivate the "Automatically merge series that are spread across multiple folders" option in the library settings. (see: https://github.com/jellyfin/jellyfin/discussions/12177)
But this will only hide them from the GUI, because the empty seasons are most likely still present in the database. And if you use this feature, deactivating it will result in multiple entries for the same shows in the library root.

**Comment by JamsRepos** [comment]
> It seems it has something to do with the "Missing Episode Fetcher" metadata provider. After deactivating it in the show library, the empty seasons don't reappear after deletion even if I trigger a rescan of the affected show.

I don’t seem to have that provider yet I get this issue. I do have “Embedded Image Extractor” and “Screen Grabber” but nothing mentioning “Missing Episode Fetcher”. 

> According to Travel2Here, it also helps to deactivate the "Automatically merge series that are spread across multiple folders" option in the library settings. (see: #12177) But this will only hide them from the GUI, because the empty seasons are most likely still present in the database. And if you use this feature, deactivating it will result in multiple entries for the same shows in the library root.

I also saw this mentioned in the Discord and I still have this ongoing issue. It’s more than likely the reason why they’re hiding as you say but because nothing has changed backend wise, they just come back after a scan.

**Comment by JamsRepos** [comment]
Just a quick update; the same logic is present even without the "Missing Episode Fetcher" or TVDB. The specials and seasons disappear on a metadata refresh and return with a scan.

**Comment by TheFeelTrain** [comment]
Disabling "Missing Episode Fetcher" makes no difference for me. They always come back after a scan.

**Comment by Aydonger** [comment]
Same here...

**Comment by fernvenue** [comment]
Seems related to https://github.com/jellyfin/jellyfin/pull/12240 this?

**Comment by zacharyPuthoff** [comment]
I'm having the exact same issue -- is there any resolution?

**Comment by felix920506** [comment]
Should be fixed by the above PR

**Comment by jkhsjdhjs** [comment]
I still experience this issue with jellyfin 10.9.8. Empty seasons disappear when replacing all metadata, but reappear after a library scan.

**Comment by fa1rid** [comment]
Yes the issue is still not fixed for me as well even after the latest update. Empty seasons still there.

**Comment by rligt** [comment]
I can also confirm that this bug is still present even with a brand new library, only way to fix it for me is reverting back to 10.9.6.
Probably should re-open this issue since the fix didn't actually fix it.

**Comment by TheFeelTrain** [comment]
Even after upgrading to 10.9.9 they are still reappearing for me.

**Comment by rligt** [comment]
Can also confirm the issue persists in 10.9.9 even when creating a brand new library.

**Comment by anthonymag** [comment]
Just spent some time with this (thanks for the folks on Discord for the guidance).

On 10.9.9:

From an individual show:
Scan for new and updated files -> adds the "shadow" seasons back
Search for missing metadata -> adds the "shadow" seasons back
Replace all metadata -> Removes the "shadow" seasons

A full library scan seems to add the shadow seasons back no matter which option I choose.

This is with or without TVBD and "Missing episode fetcher" selected.

**Comment by rligt** [comment]
I wonder if we should open a new issue for this. I'm not sure if any of the developers are aware that this is still an issue that isn't resolved.

**Comment by Brramble** [comment]
> I wonder if we should open a new issue for this. I'm not sure if any of the developers are aware that this is still an issue that isn't resolved.

Asked in the discord for a reopen.

**Comment by TimGels** [comment]
> Just spent some time with this (thanks for the folks on Discord for the guidance).
> 
> On 10.9.9:
> 
> From an individual show: Scan for new and updated files -> adds the "shadow" seasons back Search for missing metadata -> adds the "shadow" seasons back Replace all metadata -> Removes the "shadow" seasons
> 
> A full library scan seems to add the shadow seasons back no matter which option I choose.
> 
> This is with or without TVBD and "Missing episode fetcher" selected.
> 
> Edit: I also just uninstalled TVDB plugin outright and the issue still persists sadly.

I spent some time reproducing this and I noticed that once you have turned on the missing episode option it will start happening. It happened on the scans you mentioned and indeed also cleared when I scanned by replacing all metadata. 

The weird part is that after deleting the TVDB plugin and restarting Jellyfin, it kept showing this behavior. Only after I deleted the library and re-created it, it stopped doing this bug. So it seems to me that the missing episode fetch data stays lingering around somewhere, even after deleting the TVDB plugin.

So everyone here who is having this issue. Try deleting the library after you restarted Jellyfin.

**Comment by Hsclone** [comment]
Any updates on this issue? I still got a lot of empty seasons and specials and It's annoying.

**Comment by John5798** [comment]
What works for me is turning on missing episode provider in shows library and then turning on 'remove missing episodes on refresh' in the tvdb plugin and refreshing all metadata.

**Comment by arjunsrinivasan1997** [comment]
@John5798, thanks, this worked for me, but the setting I had to change was called "Include Missing Specials"

**Comment by Sunnygoesbrr** [comment]
@John5798 can confirm it works perfectly, you need to make a library refresh, if you just refresh the show itself it doesnt work

**Comment by eriksneff** [comment]
@John5798 thanks very much, this worked for me too, pretty quickly I might add. No more empty older seasons I deleted long ago!!
